assets/js/base/utils/errors.js

```diff
@@ -1,22 +1,30 @@
-const formatErrorMessage = ( error, messageProperty = 'frontendMessage' ) => {
-	if ( typeof error === 'object' && error.hasOwnProperty( 'message' ) ) {
-		return {
-			[ messageProperty ]: error.message,
-		};
-	}
-
-	return error;
-};
-
+/**
+ * Given a JS error or a fetch response error, parse and format it so it can be displayed to the user.
+ *
+ * @param {object} error Error object.
+ * @param {function} [error.json] If a json method is specified, it will try parsing the error first.
+ * @param {string} [error.message] If a message is specified, it will be shown to the user.
+ * @param {string} [error.type] The type will be used to determine how the error is shown to the user.
+ * @return {object} Error object containing a message and type.
```

Comment:
Nitpick (assuming we follow WordPress coding standards for documentation - if we don't my bad):  
- [comments should be aligned](https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/#aligning-comments)
- object notation for type should be capitalized `{Object}` (similar with array `{Array}`) as they are not primitives (whereas `string`, `number` are primitives).
- notation for function type should be capitalized `{Function}` (going by how GB is notating it).

Reply:
Thank you! We seem to be following these rules inconsistently, but agree to stick with the code standards and what Gutenberg is using.

Should be fixed in: 5e0134272c4aa1578d0662741a440b3b6a9cea16.